students/shawn/class_03/strformat_lab.py

```diff
@@ -0,0 +1,156 @@
+##String Lab: https://canvas.uw.edu/courses/1177831/assignments/4057044?module_item_id=8109753
+
+#------------------------------------------------------------------------*
+#  Task 1,2
+#------------------------------------------------------------------------*
+
+def task_one_two(val=(2, 123.4567, 10000, 12345.67)):
+
+    """Task 1,2: Format tuple into string, float, scientific with 2sig dig,scientific with 3sig dig """
+    #left pad
+    print (f"String: file_{str(val[0]).zfill(3)}")
+
+    #float
+    print(f"Float: {float(val[1])}")
+
+    #Sci 2sigdig
+    print(f"Sci 2 sigdig: {format(val[2],'.2e')}")
+
+    #Sci 3sigdig (says 3 but display 2)
+    print(f"Sci 2 sigdig: {format(val[3],'.2e')}")
+
+# task_one_two()
+
+#------------------------------------------------------------------------*
+#  Task 3
+#------------------------------------------------------------------------*
+def task_three(t=(1,2,3,4,5,6)):
+
+    """Format a tuple of ints"""
+
+    print(f"The {t.__len__()} numbers are: " + ", ".join(["{:d}"]*len(t)).format(*t))
+
+
+# task_three()
+
+#------------------------------------------------------------------------*
+#  Task 4
+#------------------------------------------------------------------------*
+def task_four(val=(4,30,2017,2,27)):
+
+    """Reorder a tuple """
+
+    reorder=[3,4,2,0,1]
+    neworder=[]
+
+    for v in reorder:
+        neworder.append(str(val[v]).zfill(2))
+    t=tuple(neworder)
+
+    print(t)
+#task_four()
+
+#------------------------------------------------------------------------*
+#  Task 5
+#------------------------------------------------------------------------*
+def get_article(word,is_singular):
+
+    """Helper function to determine appropriate preposition
+        word - [string] for evaluation
+        is_singular - [bool] indicates if function should treat as singular or plural
+    """
+
+    vowel=("a","e","i","o","u")
+    if is_singular:
+        if str(word).lower().startswith(vowel):
+            return "an"
+        else:
+            return "a"
+    else:
+        return "the"
+
```

Comment:
I like your get_article function. Was that part of the assignment and I am losing my mind, or were you just playing around? Either way, well done.